resolver-dns/src/main/java/io/netty/resolver/dns/DnsQueryContext.java

```diff
@@ -53,12 +57,15 @@
 
     DnsQueryContext(DnsNameResolver parent,
                     InetSocketAddress nameServerAddr,
-                    DnsQuestion question, Promise<AddressedEnvelope<DnsResponse, InetSocketAddress>> promise) {
-
-        this.parent = parent;
-        this.nameServerAddr = nameServerAddr;
-        this.question = question;
-        this.promise = promise;
+                    DnsQuestion question,
+                    List<DnsRecord> additional,
+                    Promise<AddressedEnvelope<DnsResponse, InetSocketAddress>> promise) {
+
+        this.parent = checkNotNull(parent, "parent");
+        this.nameServerAddr = checkNotNull(nameServerAddr, "nameServerAddr");
+        this.question = checkNotNull(question, "question");
+        this.additional = checkNotNull(additional, "additional");
```

Comment:
'List additional' is used just for building query. Only way how user can modify this list is from different thread. If someone will do it, even creating private copy don't help us. Because he can modify 'DnsRecord' or 'DnsQuestion' objects too...

Question is different, what kind of collections should be used in Netty api?